src/actionTypes.js

```diff
@@ -378,15 +401,74 @@ export type EventStreamOccupyAction = {
   streams: StreamUpdateDetails[],
 };
 
-export type EventNewMessageAction = any;
+export type EventNewMessageAction = {
+  type: typeof EVENT_NEW_MESSAGE,
+  caughtUp: CaughtUpState,
+  localMessageId: ?number,
+  message: Message,
+  ownEmail: string,
+};
+
 export type EventMessageDeleteAction = any;
 export type EventUpdateMessageAction = any;
-export type EventReactionAddAction = any;
-export type EventReactionRemoveAction = any;
-export type EventPresenceAction = any;
-export type EventTypingStartAction = any;
-export type EventTypingStopAction = any;
-export type EventUpdateMessageFlagsAction = any;
+
+export type EventReactionCommon = {
+  emoji: string,
+  messageId: number,
+  user: {
+    email: string,
+    full_name: string,
+    user_id: number,
+  },
+};
+
+export type EventReactionAddAction = EventReactionCommon & {
+  type: typeof EVENT_REACTION_ADD,
+};
+
+export type EventReactionRemoveAction = EventReactionCommon & {
+  type: typeof EVENT_REACTION_REMOVE,
+};
+
+export type EventPresenceAction = {
+  type: typeof EVENT_PRESENCE,
+  email: string,
+  presence: Presence,
+  server_timestamp: number,
+};
+
+export type EventTypingCommon = {
+  ownEmail: string,
+  recipients: Array<{
+    user_id: 104190,
```

Comment:
Wacky. Does this really only take a `user_id` of 104190? That doesn't sound right.

Reply:
Duh, nope. It is an error. Will fix.